libexec/rtld-elf/rtld_c18n.c

```diff
@@ -1636,18 +1644,32 @@ tramp_reflect(const void *data)
 #ifndef __ARM_MORELLO_PURECAP_BENCHMARK_ABI
 	data = (const char *)data - 1;
 #endif
+	/*
+	 * INVARIANT: The pointer being reflected never points to before the
+	 * entry point of the trampoline. When the fast path is enabled, the
+	 * entry point is the first instruction of the trampoline. Otherwise,
+	 * the entry point is `c18n_tramp_entry_slow_offset` bytes after the
+	 * first instruction of the trampoline.
```

Comment:
Maybe something more like:
```suggestion
	 * INVARIANT: The only pointers to trampolines that we expose are
	 * either to the entry point or to locations after it.
	 *
	 * When the fast path is enabled, the entry point is the first
	 * instruction of the trampoline. Otherwise, the entry point is
	 * `c18n_tramp_entry_slow_offset` bytes after the first instruction of
	 * the trampoline, the return point is after this different entry
	 * point, and we never expose the fast path entry point before it.
```
?

Reply:
Was dropping the final clause of this intentional?

Reply:
Oops, I didn't notice the additional text. Fixed now.